src/modules/mavlink/mavlink_messages.cpp

```diff
@@ -1996,42 +2120,60 @@ class MavlinkStreamLocalPositionNEDCOV : public MavlinkStream
 
 protected:
 	explicit MavlinkStreamLocalPositionNEDCOV(Mavlink *mavlink) : MavlinkStream(mavlink),
+		_pos_sub(_mavlink->add_orb_subscription(ORB_ID(vehicle_local_position))),
+		_pos_time(0),
 		_est_sub(_mavlink->add_orb_subscription(ORB_ID(estimator_status))),
 		_est_time(0)
 	{}
 
 	bool send(const hrt_abstime t)
 	{
-		struct estimator_status_s est = {};
+		vehicle_local_position_s pos = {};
+		estimator_status_s est = {};
 
-		if (_est_sub->update(&_est_time, &est)) {
-			mavlink_local_position_ned_cov_t msg = {};
+		bool pos_updated = _pos_sub->update(&_pos_time, &pos);
+		bool est_updated = _est_sub->update(&_est_time, &est);
 
-			msg.time_usec = est.timestamp;
-			msg.x = est.states[0];
-			msg.y = est.states[1];
-			msg.z = est.states[2];
-			msg.vx = est.states[3];
-			msg.vy = est.states[4];
-			msg.vz = est.states[5];
-			msg.ax = est.states[6];
-			msg.ay = est.states[7];
-			msg.az = est.states[8];
-
-			for (int i = 0; i < 9; i++) {
-				msg.covariance[i] = est.covariances[i];
-			}
+		mavlink_local_position_ned_cov_t pos_cov = {};
 
-			msg.covariance[9] = est.nan_flags;
-			msg.covariance[10] = est.health_flags;
-			msg.covariance[11] = est.timeout_flags;
+		if (pos_updated || est_updated) {
+			pos_cov.time_usec = est.timestamp;
 
-			mavlink_msg_local_position_ned_cov_send_struct(_mavlink->get_channel(), &msg);
+			// Current position
+			pos_cov.x = pos.x;
+			pos_cov.y = pos.y;
+			pos_cov.z = pos.z;
 
-			return true;
+			// Current linear velocity
+			pos_cov.vx = pos.vx;
+			pos_cov.vy = pos.vy;
+			pos_cov.vz = pos.vz;
+
+			// Current acceleration
+			pos_cov.ax = pos.ax;
+			pos_cov.ay = pos.ay;
+			pos_cov.az = pos.az;
+
+			// Position covariance
+			pos_cov.covariance[0] = est.covariances[7];
+			pos_cov.covariance[9] = est.covariances[8];
+			pos_cov.covariance[17] = est.covariances[9];
+
+			// Linear velocity covariance
+			pos_cov.covariance[24] = est.covariances[4];
+			pos_cov.covariance[30] = est.covariances[5];
+			pos_cov.covariance[35] = est.covariances[6];
+
+			// Accel bias covariance
```

Comment:
This is incorrect. We need acceleration covariances (which isn't actually directly estimated). Not the accelerometer bias covariances.

Reply:
I was actually uncertain of this, cause we don't really have the covariances for the accels. Is there a way we can obtain this? Probably from the IMU spec?

Reply:
Yeah we can get the std_dev from the IMU spec and calculate the variance from it. Maybe I could start by using the MPU6000 spec by default